mlrun/alerts/alert.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from mlrun.common.schemas.alert import EventEntity
+from mlrun.common.schemas.notification import Notification
+from mlrun.model import ModelObj
+
+
+class AlertConfig(ModelObj):
+    _dict_fields = [
+        "project",
+        "name",
+        "description",
+        "summary",
+        "severity",
+        "criteria",
+        "reset_policy",
+    ]
+
+    def __init__(
+        self,
+        project=None,
+        name=None,
+        description=None,
+        summary=None,
+        severity=None,
+        trigger=None,
+        criteria=None,
+        reset_policy=None,
+        notifications=None,
+        entity=None,
+        template=None,
```

Comment:
Does the `template` have to be a template object? User should be able to generate the `AlertConfig` using a template name as well (in which case we go and retrieve the template from the DB).
Also, please move the `template` to be right after the mandatory params (`project` + `name`).